### PR TITLE
Fix compat/src/internal.d.ts

### PR DESCRIPTION
**Pull request by 38elements**

### CODE REVIEW

File: test/shared/exports.test.js [hunk 1]
```diff
@@ -12,6 +12,5 @@ describe('preact', () => {
 		expect(cloneElement).to.be.a('function');
 		expect(createContext).to.be.a('function');
 		expect(options).to.exist.and.be.an('object');
-		// expect(rerender).to.be.a('function');
```
Comment (38elements) [inline test/shared/exports.test.js]: I think it is better to describe  in the release note that `rerender()` was removed.

File: hooks/src/internal.d.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export interface ComponentHooks {
 }
 
 export interface Component extends PreactComponent<any, any> {
-	__hooks: ComponentHooks;
+	__hooks?: ComponentHooks;
```
Comment (38elements) [inline hooks/src/internal.d.ts]: https://github.com/developit/preact/blob/06e5b073bd84e0c05807ce25fd2614fda618ae6b/hooks/src/index.js#L67

File: compat/src/internal.d.ts [hunk 3]
```diff
@@ -1,9 +1,30 @@
...
-export type ForwardFn<T = any, P = {}> = (props: P, ref: Ref<T>) => ComponentFactory<P>;
+export interface VNode<T = any> extends PreactVNode<T> {
+  $$typeof?: symbol | string;
+  preactCompatNormalized?: boolean;
+}
+
+export interface ForwardFn<T = any, P = {}> { 
+  (props: P, ref: Ref<T>): VNode;
+  displayName?: string;
+}
```
Comment (38elements) [inline compat/src/internal.d.ts]: https://github.com/developit/preact/blob/06e5b073bd84e0c05807ce25fd2614fda618ae6b/compat/src/index.js#L322-L331

File: compat/src/internal.d.ts [hunk 4]
```diff
@@ -1,9 +1,30 @@
...
+
+export interface FunctionalComponent<P = {}> extends PreactFunctionalComponent<P> {
+  shouldComponentUpdate?(nextProps: Readonly<P>): boolean;
+  _forwarded?: true;
 }
 
-export type ForwardFn<T = any, P = {}> = (props: P, ref: Ref<T>) => ComponentFactory<P>;
+export interface VNode<T = any> extends PreactVNode<T> {
+  $$typeof?: symbol | string;
```
Comment (38elements) [inline compat/src/internal.d.ts]: https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L335

File: compat/src/internal.d.ts [hunk 5]
```diff
@@ -1,9 +1,30 @@
...
+}
+
+export interface FunctionalComponent<P = {}> extends PreactFunctionalComponent<P> {
+  shouldComponentUpdate?(nextProps: Readonly<P>): boolean;
+  _forwarded?: true;
 }
 
-export type ForwardFn<T = any, P = {}> = (props: P, ref: Ref<T>) => ComponentFactory<P>;
+export interface VNode<T = any> extends PreactVNode<T> {
+  $$typeof: symbol | string;
```
Comment (38elements) [inline compat/src/internal.d.ts]: `options.vnode()` is executed in `createVNode()`.
https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/src/create-element.js#L66

https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L160-L165

File: compat/src/internal.d.ts [hunk 6]
```diff
@@ -1,9 +1,30 @@
...
+  preactCompatNormalized: boolean;
```
Comment (38elements) [inline compat/src/internal.d.ts]: https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L139

File: compat/src/internal.d.ts [hunk 7]
```diff
@@ -1,9 +1,30 @@
...
+export { PreactElement } from '../../src/internal';
+
+export interface Component<P = {}, S = {}> extends PreactComponent<P, S> {
+  isReactComponent: object;
+  isPureReactComponent?: true;
+}
+
+export interface FunctionalComponent<P = {}> extends PreactFunctionalComponent<P> {
+  shouldComponentUpdate?(nextProps: Readonly<P>): boolean;
+  _forwarded?: true;
```
Comment (38elements) [inline compat/src/internal.d.ts]: https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L293-L298

File: compat/src/internal.d.ts [hunk 8]
```diff
@@ -1,9 +1,30 @@
...
+} from '../../src/internal';
 
-export interface VNode<T = any> extends CoreVNode<T> {
-	$$typeof: symbol | string;
+export { ComponentChildren } from '../../src/index';
+
+export { PreactElement } from '../../src/internal';
+
+export interface Component<P = {}, S = {}> extends PreactComponent<P, S> {
+  isReactComponent: object;
```
Comment (38elements) [inline compat/src/internal.d.ts]: https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L272

File: compat/src/internal.d.ts [hunk 9]
```diff
@@ -1,9 +1,30 @@
...
+  isPureReactComponent?: true;
```
Comment (38elements) [inline compat/src/internal.d.ts]: https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L263

File: compat/src/internal.d.ts [hunk 10]
```diff
@@ -1,9 +1,30 @@
...
-export type ForwardFn<T = any, P = {}> = (props: P, ref: Ref<T>) => ComponentFactory<P>;
+export interface VNode<T = any> extends PreactVNode<T> {
+  $$typeof: symbol | string;
+  preactCompatNormalized: boolean;
+}
+
+export interface ForwardFn<P = {}, T = any> {
+  (props: P, ref: Ref<T>): VNode;
+  displayName?: string;
+}
```
Comment (38elements) [inline compat/src/internal.d.ts]: https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L322-L331

File: compat/src/internal.d.ts [hunk 11]
```diff
@@ -1,9 +1,30 @@
-import { CoreVNode } from '../../src/internal';
-export { PreactElement, ComponentFactory } from '../../src/internal'
-export { ComponentChildren, Ref, ComponentFactory, FunctionalComponent } from '../../src/index';
+{ Ref } from '../../src/index';
+{
+  Component as PreactComponent,
+  VNode as PreactVNode,
+  FunctionalComponent as PreactFunctionalComponent
+} from '../../src/internal';
```
Comment (ForsakenHarmony) [inline compat/src/internal.d.ts]: what happened here?

Comment (38elements) [reply compat/src/internal.d.ts]: This imports [Component type of Preact](https://github.com/developit/preact/blob/b84b873ee454ab545adc4259fe223e42feffdda4/src/internal.d.ts#L40) and gives an alias `PreactComponent`.

Comment (ForsakenHarmony) [reply compat/src/internal.d.ts]: but the `import` is missing

Comment (38elements) [reply compat/src/internal.d.ts]: Thank you for pointing out.
I fixed it.

File: test/shared/exports.test.js (hunk 1 above)
Comment (marvinhagemeister) [reply test/shared/exports.test.js]: Good catch, added it to the release notes under breaking changes :+1:

**Pull request merged**